### PR TITLE
Replace `set-output` usage with `GITHUB_OUTPUT`

### PR DESCRIPTION
**Pull request by parkerbxyz**
This PR is the same as #1986, which was reverted in https://github.com/oxsecurity/megalinter/commit/a47459fcb4521766738367f693b23bd98f9e164f, but with some slight changes as discussed here: https://github.com/oxsecurity/megalinter/pull/1986#discussion_r1002668154.

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [ ] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM** May be worth rebasing to consolidate the 9 commits into 2, or we can squash and merge.

**Comment by parkerbxyz** [comment]
I rebased 🙂

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM** I would probably have squashed "Consolidate redirects" into the preceding commit, but it's quite clean now!

**Comment by parkerbxyz** [comment]
> I would probably have squashed "Consolidate redirects" into the preceding commit

Oh yes, good point. Done!

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM** and thanks for adding me as a co-author!

File: .automation/build.py [hunk 1]
```diff
@@ -2757,7 +2757,9 @@ def manage_output_variables():
                 updated_versions = 1
                 break
         if updated_versions == 1:
-            print("::set-output name=has_updated_versions::1")
+            output_file = os.environ["GITHUB_OUTPUT"]
```
Comment (nvuillam) [inline .automation/build.py:2760]: If we are not on github (like when building locally) the variable will be empty, the code has to manage such case ^^ ( if ^^ )

File: megalinter/MegaLinter.py [hunk 2]
```diff
@@ -822,7 +822,9 @@ def display_header():
         logging.info("")
 
     def check_results(self):
-        print(f"::set-output name=has_updated_sources::{str(self.has_updated_sources)}")
+        output_file = os.environ["GITHUB_OUTPUT"]
```
Comment (nvuillam) [inline megalinter/MegaLinter.py:825]: check for existence of GITHUB_OUTPUT presence ^^ ( in case we are running in gitlab for example ^^ )

**Review by nvuillam — CHANGES_REQUESTED** [review]
plz check comments :)

File: .automation/build.py (hunk 1 above)
Comment (parkerbxyz) [reply .automation/build.py:2760]: What should happen when not running on GitHub? Print to console like it does now?

Comment (Kurt-von-Laven) [reply .automation/build.py:2760]: In my opinion, nothing at all. The output environment file and deprecated `set-output` command are GitHub Actions concepts.

**Comment by Kurt-von-Laven** [comment]
I suppose "Double quote to prevent globbing and word splitting" can be squashed into the preceding commit as well.

**Review by nvuillam — CHANGES_REQUESTED** [review]
Plz check comments :)

File: megalinter/MegaLinter.py [hunk 3]
```diff
@@ -822,7 +822,10 @@ def display_header():
         logging.info("")
 
     def check_results(self):
-        print(f"::set-output name=has_updated_sources::{str(self.has_updated_sources)}")
+        github_output_file = os.environ["GITHUB_OUTPUT"]
```
[the remark this answers is not in the document]
Comment (parkerbxyz) [reply megalinter/MegaLinter.py]: Done: https://github.com/oxsecurity/megalinter/pull/1996/commits/db4898c2192c55866b8e7d0265259a78db576c65

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ Note: Can be used with `oxsecurity/megalinter@beta` in your GitHub Action mega-l
   - [pylint](https://pylint.pycqa.org) from 2.15.5 to **2.12.2** on 2022-10-24
   - [rubocop](https://rubocop.org/) from 1.37.0 to **1.37.1** on 2022-10-24
   - [yamllint](https://yamllint.readthedocs.io/) from 1.28.0 to **1.26.3** on 2022-10-24
+- Replace `set-output` usage with `GITHUB_OUTPUT`
```
[the remark this answers is not in the document]
Comment (parkerbxyz) [reply CHANGELOG.md]: Done: https://github.com/oxsecurity/megalinter/pull/1996/commits/47ca91290aca891d100e21b5e0dbaad12127875d

File: .automation/build.py [hunk 5]
```diff
@@ -2757,7 +2757,10 @@ def manage_output_variables():
                 updated_versions = 1
                 break
         if updated_versions == 1:
-            print("::set-output name=has_updated_versions::1")
+            github_output_file = os.environ["GITHUB_OUTPUT"]
```
[the remark this answers is not in the document]
Comment (parkerbxyz) [reply .automation/build.py]: Done: https://github.com/oxsecurity/megalinter/pull/1996/commits/db4898c2192c55866b8e7d0265259a78db576c65

**Comment by nvuillam** [comment]
Seems nice, let's see what CI says :)

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM**

**Comment by llaville** [comment]
Thanks to all for your works.
I've a little question : is it possible to build also a v5 release with these changes ?

**Review by nvuillam — APPROVED** [review]
Thanks a lot !
Let's try it with beta version :)

**Comment by nvuillam** [comment]
> Thanks to all for your works. I've a little question : is it possible to build also a v5 release with these changes ?

I'm sorry, v5 is not maintained since v6 and migration to ox security organization

Do you have a blocking issue with v6 ?

**Pull request merged**